src/MAT_HDF5.jl

```diff
@@ -62,7 +62,11 @@ function close(f::MatlabHDF5File)
         if f.writeheader
             magic = zeros(UInt8, 512)
             identifier = "MATLAB 7.3 MAT-file" # minimal but sufficient
-            magic[1:length(identifier)] = Vector{UInt8}(identifier)
+            GC.@preserve magic identifier begin
```

Comment:
This has nothing to do with this PR so you don't have to change anything but I feel like `magic` should just be a global constant array here...